### PR TITLE
nim: update to 1.6.8

### PR DESCRIPTION
**Pull request by esafak**
###### Type(s)

- [x] update
- [ ] bugfix
- [ ] enhancement
- [ ] security fix

@gpanders I deleted the patch file because it needs to be updated with every release, is not strictly needed, and is better addressed at the source: https://github.com/nim-lang/Nim/blob/devel/tools/niminst/install.nimf

###### Tested on

macOS 12.6
M1 Mac mini

```lang=bash
> sudo port install nim
--->  Computing dependencies for nim
--->  Fetching archive for nim
--->  Attempting to fetch nim-1.6.8_0.darwin_21.arm64.tbz2 from https://packages.macports.org/nim
--->  Attempting to fetch nim-1.6.8_0.darwin_21.arm64.tbz2 from https://ywg.ca.packages.macports.org/mirror/macports/packages/nim
--->  Attempting to fetch nim-1.6.8_0.darwin_21.arm64.tbz2 from https://kmq.jp.packages.macports.org/nim
--->  Fetching distfiles for nim
--->  Verifying checksums for nim
--->  Extracting nim
--->  Configuring nim
--->  Building nim
--->  Staging nim into destroot
--->  Installing nim @1.6.8_0
--->  Deactivating nim @1.6.4_0
--->  Cleaning nim
--->  Activating nim @1.6.8_0
--->  Cleaning nim
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.

> nim -v
Nim Compiler Version 1.6.8 [MacOSX: arm64]
Compiled at 2022-10-05
Copyright (c) 2006-2021 by Andreas Rumpf

active boot switches: -d:release
```

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by gpanders** [comment]
>@gpanders I deleted the patch file because it needs to be updated with every release, is not strictly needed, and is better addressed at the source: https://github.com/nim-lang/Nim/blob/devel/tools/niminst/install.nimf

@tuffnatty Is the one who added it, so I'll ping them.

**Comment by barracuda156** [comment]
Cool. I have WIP on `Nim` for PPC against upstream branch, so 1.6.8 will probably make subsequent PR easier.

**Comment by esafak** [comment]
> > @gpanders I deleted the patch file because it needs to be updated with every release, is not strictly needed, and is better addressed at the source: https://github.com/nim-lang/Nim/blob/devel/tools/niminst/install.nimf
> 
> @tuffnatty Is the one who added it, so I'll ping them.

I might add that quoting has already been implemented upstream: nim-lang/Nim#20051

File: lang/nim/Portfile [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 name                nim
-version             1.6.4
+version             1.6.8
 revision            0
 license             MIT
 categories          lang
 platforms           darwin
 supported_archs     i386 x86_64 ppc64 arm64
-maintainers         {@gpanders gpanders.com:greg} \
+maintainers         {@gpanders gpanders.com:greg} {@esafak gmail.com:esafak} \
```
Comment (gpanders) [inline lang/nim/Portfile:12]: @esafak You can remove me as a maintainer as part of this PR as well.

**Review by gpanders — APPROVED** [review]

**Comment by barracuda156** [comment]
@gpanders @esafak By the way, maybe you could take a look and suggest me a solution for this error?
https://github.com/nim-lang/Nim/issues/19914#issuecomment-1263509043 (identical error on 10.6.8 Rosetta and 10.6 PPC).

**Pull request merged**